python/google/protobuf/internal/json_format_test.py

```diff
@@ -909,6 +909,26 @@ def testParseEnumValue(self):
         'for enum type protobuf_unittest.TestAllTypes.NestedEnum.',
         json_format.Parse, '{"optionalNestedEnum": 12345}', message)
 
+  def testParseUnknownEnumStringValueProto3(self):
+    message = json_format_proto3_pb2.TestMessage()
+    text = '{"enumValue": "UNKNOWN_STRING_VALUE"}'
+    json_format.Parse(text, message, ignore_unknown_fields=True)
+    # In proto3, there is no difference between the default value and 0.
+    self.assertEqual(message.enum_value, 0)
+
+  def testParseUnknownEnumStringValueProto2(self):
+    message = json_format_pb2.TestNumbers()
+    text = '{"a": "UNKNOWN_STRING_VALUE"}'
```

Comment:
curious why the key is `a` here instead of `enumValue` like in other tests?

Reply:
Good catch, I verified that this is OK and added comment.

```
    # Note that in src/google/protobuf/util/json_format.proto::TestNumbers
    # has a field 'optional MyType a = 1', which is different
    # from the test below that are using
    # src/google/protobuf/util/json_format_proto3.proto::TestMessage.
```